lib/room.js

```diff
@@ -509,6 +546,16 @@ function connectParticipant(room, participantSignaling) {
   });
 }
 
+function handleLocalParticipantEvents(room, localParticipant) {
+  localParticipant.on('trackWarning', (twilioWarningName, localTrack) => {
+    room.emit('trackWarning', twilioWarningName, localTrack, localParticipant);
+  });
+
+  localParticipant.on('trackWarningsCleared', localTrack => {
+    room.emit('trackWarningsCleared', localTrack, localParticipant);
+  });
```

Comment:
```suggestion
  const events =  [
    'trackWarning',
    'trackWarningsCleared'
  ];
  const reemitEvent = (...args) =>
    room.emit(event, ...[...args, participant]);

  events.forEach(event =>
    localParticipant.on(event, reemitEvent));

  room.once('disconnected', () => events.forEach(event =>
      localParticipant.removeListener(event, reemitEvent)));
```

Reply:
I understand what you're trying to do, but the suggested code doesn't work as is. I made some tweaks to make it work. Please see latest.